pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -549,31 +549,51 @@ public final synchronized void readEntriesComplete(List<Entry> entries, Object c
             log.debug("[{}] Distributing {} messages to {} consumers", name, entries.size(), consumerList.size());
         }
 
+       sendMessagesToConsumers(readType, entries);
+    }
+
+    protected synchronized void sendInProgressAcquire() {
+        sendInProgress = true;
+    }
+
+    protected synchronized void sendInProgressRelease() {
+        sendInProgress = false;
+    }
+
+    protected synchronized boolean sendInProgress() {
+        return sendInProgress;
+    }
+
+    protected final synchronized void sendMessagesToConsumers(ReadType readType, List<Entry> entries) {
+        java.util.function.Consumer<Boolean> consumer = asyncRead -> {
```

Comment:
Is it possible to avoid creating a new `java.util.function.Consumer` for each consumer's message dispatching?